src/runtime_src/core/tools/xbtracer/src/lib/detail/windows/logger.h

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <iostream>
+
+namespace xrt::tools::xbtracer {
+
+constexpr std::string_view path_separator = "\\";
+
+/*
+ * Wrapper api for time formating
+ * */
+inline std::tm localtime_xp(std::time_t timer)
+{
+  std::tm bt{};
+  localtime_s(&bt, &timer);
+  return bt;
+}
+
+std::string get_env(std::string key)
+{
+  char *val = nullptr;
+  size_t len = 0;
+  errno_t err = _dupenv_s(&val, &len, key.c_str());
+  if (err != 0 || val == nullptr)
+    return std::string();
+  std::string result(val);
+  free(val); // Free the memory allocated by _dupenv_s
+  return result;
+}
+
+std::string wide_to_string(const std::wstring& wstr)
+{
+  if (wstr.empty()) return std::string();
```

Comment:
Always put stmt on new line otherwise debugging and setting breaks points is difficult,
```suggestion
  if (wstr.empty()) 
    return std::string();
```

Reply:
agree, updated as suggested.